File: 004. Find Duplicate In Array/findDuplicateInArray.java
import java.util.Arrays;

public class findDuplicateInArray {
    public static void main(String[] args) {
        int[] nums = {1, 3, 4, 2, 2};
        System.out.println(findDuplicate(nums));
    }
    public static int findDuplicate(int[] nums) {
        int numsLength = nums.length;
        Arrays.sort(nums);
        int i=0;
        while(i<numsLength){
            if(nums[i]==nums[i+1]){
                return nums[i];
            }
            i++;
        }
        return 0;
    }
}

File: 004. Find Duplicate In Array/findDuplicateInArray.cpp
#include <bits/stdc++.h>
using namespace std;

int findDuplicate(vector<int> &arr, int n){
	sort(arr.begin(), arr.end());
	int i=0;
	while(i<n){
		if(arr[i]==arr[i+1]){
			return arr[i];
		}
		i++;
	}
}

int main() {
    vector<int> nums = {1, 3, 4, 2, 2};
    
    cout<<findDuplicate(&nums, 5);
    
    return 0;
}